tests/network-tests/src/nicaea/dto/workingGroupOpening.ts

```diff
@@ -183,10 +182,12 @@ export class WorkingGroupOpening {
     this.openingType = value;
   }
 
-  constructor() {}
+  constructor() {
```

Comment:
Going to disagree with that point- one of the reasons why we have introduced a data transfer object is to mitigate the risk of passing the values in the wrong order: https://github.com/Joystream/joystream/pull/782#discussion_r447783424

If all the values will be passed to the constructor, that goal will not be achieved- instead of risk having the wrong order in the function call, we will have the same risk but for the constructor call.

Reply:
Yes as in the original comment you linked to I wasn't implying re-introducing a long list of arguments. But this approach here of depending on methods might leave some of the private fields un-initialized?

Reply:
Yes, it might. Normally typescript won't allow that, that is why assignment assertion (exclamation sign) is used. What is the best approach you can suggest?

Reply:
Yes so to this point exactly, I guess the approach here mainly solves problem of passing arguments in wrong order.
So passing an object was the correct solution, except now this new object can have null or underfined fields. So wouldn't simply passing arguments like so be the optimal approach:

```javascript
// original approach which I advised against
function withHugeArgsList(a, b, c, ...., z) {

}

// suggested method, takes a single object instead
function theMethod({ arg_a: A, arg_b: B, arg_c: C }) {
   do_something_with_args()
}

// calling the method
theMethod({arg_a: valueOfTypeA, arg_b: valueOfTypeB, ...})
```

This way will insure no incorrect ordering of arguments, and that all args are provided (optionally give some args a default value if you want them to be optional)

Reply:
Ok, will do.